src/Libraries/Nop.Services/Customers/Caching/CustomerCacheEventConsumer.cs

```diff
@@ -23,6 +23,23 @@ public async Task HandleEventAsync(CustomerPasswordChangedEvent eventMessage)
             await RemoveAsync(NopCustomerServicesDefaults.CustomerPasswordLifetimeCacheKey, eventMessage.Password.CustomerId);
         }
 
+        /// <summary>
+        /// Clear cache by entity event type
+        /// </summary>
+        /// <param name="entity">Entity</param>
+        /// <param name="entityEventType">Entity event type</param>
+        /// <returns>A task that represents the asynchronous operation</returns>
+        protected override async Task ClearCacheAsync(Customer entity, EntityEventType entityEventType)
+        {
+            if (entityEventType == EntityEventType.Delete)
+            {
+                await RemoveAsync(NopCustomerServicesDefaults.CustomerAddressesCacheKey, entity);
+                await RemoveByPrefixAsync(NopCustomerServicesDefaults.CustomerAddressesByCustomerPrefix, entity);
```

Comment:
For this code to work properly, you must also change the CustomerAddressCacheKey key, since it is now erroneous

Reply:
Yes that's right

Reply:
Fixed